src/result.rs

```diff
@@ -77,7 +77,7 @@ pub struct QueryResult<'a> {
 
 impl<'a> QueryResult<'a> {
     pub(crate) fn new(
-        token_stream: Pin<Box<dyn Stream<Item = crate::Result<ReceivedToken>> + 'a>>,
+        token_stream: Pin<Box<dyn Stream<Item = crate::Result<ReceivedToken>> + Send + Sync + 'a>>,
```

Comment:
Ah interesting; first time I'm seeing a [`BoxStream`](https://docs.rs/futures/0.3.5/futures/stream/type.BoxStream.html) equivalent .

Reply:
Actually wondering if perhaps pulling this in as an alias would be useful. Seeing `BoxStream` can help with readability somewhat. The `Sync` bound isn't part of it tho, but I'm assuming that's not required? -- Borrowing a Box for use inside of another task sounds like an unlikely thing to want.

Reply:
Oh, wow, of course there is `BoxStream`. I think we have a redundant `Sync` boundary in Quaint that forced me to do this here...

Reply:
Actually, I know of exactly one time where Sync was needed for something that's *possibly* similar: https://github.com/http-rs/surf/issues/97. Could be that it's actually needed in quaint, but in that case it could still be useful to define a `BoxStream` alias inside the crate that has the extra bound

Reply:
Yeah, we kind of need sync in quaint, but not for the stream it seems. For `ToSql` it is needed tho.